plugins/rsync_server.py

```diff
@@ -0,0 +1,157 @@
+
+        self.server = server
+        self.local_path = local_path
+        self.server_path = server_path
+        self.username = username
+        self.update_time = update_time
+        self.include = include
+        self.time_key = time_key
+
+    def execute(self, input_data=None):
+        """
+        Executing an action upon detection an arbitrary condition in the data.
+
+        Parameters
+        ----------
+        input_data : any, optional
+            Per iteration input data passed to this function from previous
+            PipelineSteps. Not used here but retained for compatibility with
+            the generalized PipelineStep API. The default is None.
+
+        Returns
+        -------
+        input_data : same as input_data
+            Input data passed through, for further steps to consume. Note
+            a new key will be added. See documentation for initialization
+            of the class.
+        """
+
+        # Handle first iteration
+        if self._previous_data is None:
+            input_data[self.time_key] = input_data['time']
+            self._previous_data = input_data
+
+        try:
+            # How long since we last updated?
+            dt = input_data['time'].value-self._previous_data[self.time_key].value
+
+            if dt.total_seconds() > self.update_time:
+                # Now, sync the files to the server
+                out = self.rsync()
+                self.logger.info(out)
+
+                # Update the last run time before leaving
+                self._previous_data[self.time_key].value = input_data['time'].value
+
+        # If expression evaluation fails, presumably due to bad data values
+        except Exception as e:
+            self.logger.error(
+                "%s evaluating in %s on step %s: %s",
+                type(e).__name__, type(self), self.name, e)
+            self.logger.info("Error details:", exc_info=True)
+            for pretty_name, data in [("Current", input_data),
+                                      ]:
+                self.logger.info(
+                    "%s data: %r", pretty_name,
+                    {key: str(value) for key, value in data.items()})
+
+            # todo: also a status bot error message?
+
+        # Passthrough the input for consumption by any further steps
+        return input_data
+
+    def rsync(self):
+        """ Build the command for rsync and run it."""
+
+        TIMEOUT = 10  # todo: Find the best value for this and/or make it an arg
+
+        cmd = [
+            "rsync",
+            "-avz",
+        ]
+
+        # If we only include certain files, put that in there,
+        # but be careful with quotes - bash and python syntax is not the same!
+        if self.include:
+            cmd = cmd + [
+                f"--include={self.include}",
+                "--exclude=*",
+            ]
+
+        # Finish building the command for rsync
+        cmd = cmd + [
+            f"{self.local_path}",
```

Comment:
```suggestion
            f"{self.local_path.as_posix()}",
```

Convert back to string from Path for rsync